crm_compassion/models/interaction_resume.py

```diff
@@ -58,106 +58,117 @@ def populate_resume(self, partner_id):
-                FROM "crm_phonecall" as crmpc
-                JOIN res_partner p ON crmpc.partner_id = p.id
-                WHERE (p.contact_id = %s OR p.id = %s)
-                )
-    -- outgoing e-mails
-            UNION (
-              SELECT
-                'Email' as communication_type,
-                m.date as communication_date,
-                COALESCE(p.contact_id, p.id) AS partner_id,
-                mt.recipient_address as email,
-                COALESCE(config.name, m.subject) as subject,
-                REGEXP_REPLACE(m.body, '<img[^>]*>', '') AS body,
-                'out' AS direction,
-                0 as phone_id,
-                mail.id as email_id,
-                0 as message_id,
-                job.id as paper_id,
-                COALESCE(mt.state, 'error') as tracking_status
-                FROM "mail_mail" as mail
-                JOIN mail_message m ON mail.mail_message_id = m.id
-                JOIN mail_mail_res_partner_rel rel
-                ON rel.mail_mail_id = mail.id
-                FULL OUTER JOIN mail_tracking_email mt ON mail.id = mt.mail_id
-                JOIN res_partner p ON rel.res_partner_id = p.id
-                FULL OUTER JOIN partner_communication_job job
-                    ON job.email_id = mail.id
-                FULL OUTER JOIN partner_communication_config config
-                    ON job.config_id = config.id
-                    AND config.name != 'Default communication'
-                WHERE mail.state = ANY (ARRAY ['sent', 'received'])
-                AND (p.contact_id = %s OR p.id = %s)
-                )
-    -- incoming messages from partners
-            UNION (
-              SELECT
-                'Email' as communication_type,
-                m.date as communication_date,
-                COALESCE(p.contact_id, p.id) AS partner_id,
-                m.email_from as email,
-                m.subject as subject,
-                REGEXP_REPLACE(m.body, '<img[^>]*>', '') AS body,
-                'in' AS direction,
-                0 as phone_id,
-                0 as email_id,
-                m.id as message_id,
-                0 as paper_id,
-                NULL as tracking_status
-                FROM "mail_message" as m
-                JOIN res_partner p ON m.author_id = p.id
-                WHERE m.subject IS NOT NULL
-                AND m.message_type = 'email'
-                AND (p.contact_id = %s OR p.id = %s)
-                )
-                """, [partner_id] * 8)
-        for row in self.env.cr.dictfetchall():
-            # ensure that "Example <ex@exmaple.com>" is correctly printed
-            row['email'] = html_sanitize(row['email'])
-            self.create(row)
+        original_partner = self.env['res.partner'].search([
+            ('id', '=', partner_id)
+        ])
+        email_address = original_partner.email
+        partners_with_same_email_ids = self.env['res.partner'].search([
+            ('email', '=', email_address)
```

Comment:
We don't want to group the partners that don't have email address together!
```suggestion
            ('email', '=', email_address),
            ('email', '!=', False)
```

Reply:
Please restore this fix.